watertap/examples/flowsheets/case_studies/activated_sludge/ASM1_flowsheet_seq.py

```diff
@@ -0,0 +1,422 @@
+    m.fs.R1.inlet.conc_mass_comp[0, "X_I"].fix(1149 * pyo.units.g / pyo.units.m**3)
+    m.fs.R1.inlet.conc_mass_comp[0, "X_S"].fix(89.324 * pyo.units.g / pyo.units.m**3)
+    m.fs.R1.inlet.conc_mass_comp[0, "X_BH"].fix(
+        2542.03 * pyo.units.g / pyo.units.m**3
+    )
+    m.fs.R1.inlet.conc_mass_comp[0, "X_BA"].fix(148.6 * pyo.units.g / pyo.units.m**3)
+    m.fs.R1.inlet.conc_mass_comp[0, "X_P"].fix(448 * pyo.units.g / pyo.units.m**3)
+    m.fs.R1.inlet.conc_mass_comp[0, "S_O"].fix(0.3928 * pyo.units.g / pyo.units.m**3)
+    m.fs.R1.inlet.conc_mass_comp[0, "S_NO"].fix(8.32 * pyo.units.g / pyo.units.m**3)
+    m.fs.R1.inlet.conc_mass_comp[0, "S_NH"].fix(7.696 * pyo.units.g / pyo.units.m**3)
+    m.fs.R1.inlet.conc_mass_comp[0, "S_ND"].fix(1.9404 * pyo.units.g / pyo.units.m**3)
+    m.fs.R1.inlet.conc_mass_comp[0, "X_ND"].fix(5.616 * pyo.units.g / pyo.units.m**3)
+    m.fs.R1.inlet.alkalinity.fix(4.704 * pyo.units.mol / pyo.units.m**3)
+    m.fs.R1.volume.fix(1000 * pyo.units.m**3)
+    m.fs.R1.initialize()
+
+    solve_fs(m)
+
+    # Second reactor (anoxic) - standard CSTR
+    m.fs.R2 = CSTR(
+        default={
+            "property_package": m.fs.props,
+            "reaction_package": m.fs.rxn_props,
+        }
+    )
+    m.fs.R2.volume.fix(1000 * pyo.units.m**3)
+    m.fs.stream3 = Arc(source=m.fs.R1.outlet, destination=m.fs.R2.inlet)
+    pyo.TransformationFactory("network.expand_arcs").apply_to(m)
+    propagate_state(m.fs.stream3)
+    m.fs.R2.initialize()
+
+    results = solve_fs(m)
+
+    # Third reactor (aerobic) - CSTR with injection
+    m.fs.R3 = CSTR_Injection(
+        default={
+            "property_package": m.fs.props,
+            "reaction_package": m.fs.rxn_props,
+        }
+    )
+    m.fs.R3.volume.fix(1333 * pyo.units.m**3)
+    m.fs.R3.injection[...].fix(0)
+    m.fs.R3.injection[:, "Liq", "S_O"].fix(0.022)
+    m.fs.stream4 = Arc(source=m.fs.R2.outlet, destination=m.fs.R3.inlet)
+    pyo.TransformationFactory("network.expand_arcs").apply_to(m)
+    propagate_state(m.fs.stream4)
+    m.fs.R3.initialize()
+
+    results = solve_fs(m)
+
+    # Fourth reactor (aerobic) - CSTR with injection
+    m.fs.R4 = CSTR_Injection(
+        default={
+            "property_package": m.fs.props,
+            "reaction_package": m.fs.rxn_props,
+        }
+    )
+    m.fs.R4.volume.fix(1333 * pyo.units.m**3)
+    m.fs.R4.injection[...].fix(0)
+    m.fs.R4.injection[:, "Liq", "S_O"].fix(0.021)
+    m.fs.stream5 = Arc(source=m.fs.R3.outlet, destination=m.fs.R4.inlet)
+    pyo.TransformationFactory("network.expand_arcs").apply_to(m)
+    propagate_state(m.fs.stream5)
+    m.fs.R4.initialize()
+
+    # Fifth reactor (aerobic) - CSTR with injection
+    m.fs.R5 = CSTR_Injection(
+        default={
+            "property_package": m.fs.props,
+            "reaction_package": m.fs.rxn_props,
+        }
+    )
+    m.fs.R5.volume.fix(1333 * pyo.units.m**3)
+    m.fs.R5.injection[...].fix(0)
```

Comment:
```suggestion
    m.fs.R5.injection[...].fix(0)
```
What does the [...] do? Is it sort of like using (:) in MATLAB, where this would grab all indices?

Reply:
Yes, it is a slicer, but `:` and `...` have different meanings. `:` means all elements of a single index, `...` means all elements of all indices (i.e. `:, :, ...`).

Reply:
I see. That was what I meant by my MATLAB example, since using (:) in MATLAB gives you all elements of all indices.